### DISCUSSION TITLE
Improvements to godot built-in code editor

### DISCUSSION THREAD

**Discussion by vaguesyntax** [godotengine/godot-proposals#10110] [Editor]
### Describe the project you are working on

A 3D game of surviving from zombies , crafting , base building and more. Like a 7 Days To Die

### Describe the problem or limitation you are having in your project

While making a complicated project scripts are always messy. Even I use  regions more effectively they are still looks messy

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Maybe we can divide the script into several parts with the visual screen such as the screen where we choose the scripts. Like building, crafting , surviving elements parts. Because dealing with regions (probably only for me) is boring and doesn't look very good visually. And most of the time, the #endregion line is visible even if we have closed the region.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

As I mentioned above, if there is a script selection screen or a screen that looks better to the eye, the problem will be solved very well.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

If this feature is not used, there will be no problem in terms of code because there is no problem in terms of code.

### Is there a reason why this should be core and not an add-on in the asset library?

It doesn't necessarily need to be added to godot, a simple plugin that works stably and doesn't break our code can also do the trick.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concret technical implementation provided.

Could you provide some mockups of what this functionality would look like? It's unclear to me what "visual screen" refers to.

**Reply by vaguesyntax** [reply]
https://github.com/godotengine/godot-proposals/discussions/10110#discussioncomment-9960978
IDK how to use github :( . İnstead of reply i made a comment to discussion. My bad.

**Comment by vaguesyntax** [comment]
![ide](https://github.com/godotengine/godot-proposals/assets/173799252/59e1383b-c034-483e-afcb-684d3b61d5b3)
I think something like this, but of course it could be better. In the simplest way, the regions in the script can appear on the left, and when we select a region from the left, we only see the codes in that region, and thus complex project management can be easier.By the way, I apologize for opening this discussion to the issues section.
